libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Conway/DeltaDeposit.hs

```diff
@@ -0,0 +1,250 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- | The Spec and the STS rules differ on where Deposits for Certificates are
+--   recorded in the Spec.  In the Spec Deposits are recorded in the UTXO rule
+--   in the STS rules (in ConwayEra) they are recorded in the "CERT rule.
+--   This module helps track those differences.
+module Test.Cardano.Ledger.Constrained.Conway.DeltaDeposit where
+
+import Cardano.Ledger.BaseTypes (Inject (..), Network)
+import Cardano.Ledger.CertState
+import Cardano.Ledger.Coin (Coin (..))
+import Cardano.Ledger.Conway.Governance (GovActionId (..), VotingProcedures)
+import Cardano.Ledger.Conway.Rules (CertEnv (..), CertsEnv (..), ConwayDelegEnv (..))
+import Cardano.Ledger.Conway.TxCert
+import Cardano.Ledger.Core
+import Cardano.Ledger.Credential (Credential (..))
+import Cardano.Ledger.Crypto (Crypto, StandardCrypto)
+import Cardano.Ledger.Keys (KeyHash (..), KeyRole (..))
+import Cardano.Ledger.PoolParams (PoolParams (..))
+import Cardano.Ledger.UMap (CompactForm (CompactCoin), RDPair (..))
+import qualified Cardano.Ledger.UMap as UMap
+import Constrained (HasSimpleRep (..), HasSpec (..))
+import Control.DeepSeq (NFData)
+import Data.Foldable (Foldable (..))
+import qualified Data.Map as Map
+import Data.Map.Strict (Map)
+import Data.Maybe.Strict (StrictMaybe (..))
+import Data.Sequence (Seq)
+import GHC.Generics (Generic)
+import Lens.Micro
+import Test.Cardano.Ledger.Constrained.Conway.Instances (IsConwayUniv)
+import Test.Cardano.Ledger.Conway.TreeDiff (ToExpr)
+import Test.QuickCheck (Gen)
+
+-- ==============================================================
+
+-- A Sum type of all the different ways Deposits are recorded.
+-- In the STS rules each kind of deposit is stored in map with
+-- Credential or KeyHash or GovActionId as the key, Using this type
+-- we can use a single Map with DepositPurpose as the key. This is
+-- how the Spec does it, rather than in 4 maps in different parts of the Spec.
+data DepositPurpose c
+  = CredentialDeposit !(Credential 'Staking c)
+  | PoolDeposit !(KeyHash 'StakePool c)
+  | DRepDeposit !(Credential 'DRepRole c)
+  | GovActionDeposit !(GovActionId c)
+  deriving (Generic, Eq, Show, Ord)
+
+instance ToExpr (DepositPurpose c)
+instance Crypto c => NFData (DepositPurpose c)
+instance HasSimpleRep (DepositPurpose c)
+instance (IsConwayUniv fn, Crypto c) => HasSpec fn (DepositPurpose c)
+
+-- ======================================================================================
+
+-- Deposits are recorded when the deposit is made (registering) and refunded (deregistering)
+data DepositKind = MakeDeposit Coin | RefundDeposit Coin deriving (Show)
```

Comment:
I think the name might be a bit confusing because of the meaning that the word "kind" has in Haskell. Maybe a better name would be something like `DepositChange`

Reply:
`DepositAction` is also an option